producer/logger.go

```diff
@@ -27,7 +27,9 @@ func logConfig(producerConfig *ProducerConfig) log.Logger {
 		}
 	}
 	var logger log.Logger
-	if producerConfig.IsJsonType {
+	if producerConfig.Logger != nil {
```

Comment:
If custom logger is not nil, maybe we can just skip logConfig?
```
var logger log.Logger
if producerConfig.Logger != nil {
     logger = producerConfig.Logger
} else {
    logger = logConfig(producerConfig)
}
```

Reply:
But we need the level filter in logConfig.

Reply:
That's ok, you could wrapper it yourself before passing to producerConfig, `NewFilter` also returns a new logger.
You can do anything you like
```
yourLogger := log.NewJSONLogger(writer)
yourLogger = level.NewFilter(yourLogger, level.AllowDebug())
yourLogger = log.With(yourLogger, "time", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
yourLogger = log.With(yourLogger, "hello", "this is my machine 1.1.1.1")
```

Then you can pass it to producerConfig
```
producerConfig := GetDefaultProducerConfig()
producerConfig.Logger = yourLogger
```

Reply:
Instead, if we modify the logger, it may override your settings unexpectedly